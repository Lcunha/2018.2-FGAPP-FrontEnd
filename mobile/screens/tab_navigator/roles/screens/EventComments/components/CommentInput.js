import React, { Component } from 'react';
import {
    Text,
    ToastAndroid,
    StyleSheet,
    TextInput,
    Alert,
    ScrollView
} from 'react-native';
import { Button, Toast } from 'native-base';
import PropTypes from 'prop-types';
import moment from 'moment';
import Divider from '../../EventProfile/components/Divider';
import { ACTION_USER_DICTIONARY_INSERT } from 'expo/src/IntentLauncherAndroid';

class CommentInput extends Component {
    state = {
        author: '',
        text: '',
        answerId: null,
        created: null,
        edited: null,
        eventId: 0,
        responseStatus: 0,
        responseOk: false
    };

    _commentValidate() {
        if (this.state.author.length <= 1) {
            Alert.alert(
                'Erro ao postar o comentário',
                'Nome do Autor deve ter pelo menos 2 caracteres.'
            );
            return false;
        } else if (this.state.text.length <= 1) {
            Alert.alert(
                'Erro ao postar o comentário',
                'O comentário deve ter pelo menos 2 caracteres.'
            );
            return false;
        } else return true;
    }

    _postComment = async () => {
        if (this._commentValidate() == false)
            ToastAndroid.show(
                'Erro ao postar o comentário.',
                ToastAndroid.LONG
            );
        else {
            const { eventId, userId} = this.props;
            const {
                text,
                edited,
            } = this.state;

            fetch('http://roles-comments.herokuapp.com/comment/', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    authorName: this.props.userInfo.name,
                    authorId: userId,
                    text,
                    answerId: 0,
                    created: moment().format('YYYY-MM-DD HH:mm:ss'),
                    edited,
                    eventId: eventId
                })
            })
                .then(response => {
                    this.setState({
                        responseStatus: response.status,
                        responseOk: response.ok
                    });
                    response.json();
                })
                .then(responseJson => {
                    // console.log(
                    //     `ResponseOk: ${responseOk}  //  Status: ${responseStatus}`
                    // );
                    if (
                        this.state.responseStatus >= 200 &&
                        this.state.responseStatus <= 202 &&
                        this.state.responseOk
                    ) {
                        // console.log(
                        //     'Comentário postado com sucesso!\n',
                        //     responseJson
                        // );
                        ToastAndroid.show(
                            'Comentário postado com sucesso!',
                            ToastAndroid.SHORT
                        );
                        this._textArea.clear();
                        this.props.onSubmit();
                    } else {
                        return Promise.reject();
                    }
                })
                .catch(() => {
                    Alert.alert(
                        'Erro ' + this.state.responseStatus,
                        'Houveram problemas na conexão. Tente novamente mais tarde.'
                    );
                });
        }
    };

    render() {
        return (
            <ScrollView keyboardShouldPersistTaps="always">
                <TextInput
                    style={styles.textAreaAuthor}
                    placeholder="Autor"
                    underlineColorAndroid="transparent"
                    multiline={false}
                    onChangeText={text => this.setState({ author: text })}
                    maxLength={40}
                />

                <TextInput
                    ref={input => {
                        this._textArea = input;
                    }}
                    style={styles.textAreaComment}
                    placeholder="Novo Comentário"
                    underlineColorAndroid="transparent"
                    multiline={false}
                    onChangeText={text => this.setState({ text: text })}
                    maxLength={128}
                />
                <Button
                    success
                    block
                    style={styles.submitCommentBtn}
                    onPress={() => this._postComment()}
                >
                    <Text>Enviar Comentário</Text>
                </Button>
                <Divider size="80%" />
            </ScrollView>
        );
    }
}
export default CommentInput;

CommentInput.propTypes = {
    eventId: PropTypes.node.isRequired
};

const styles = StyleSheet.create({
    textAreaAuthor: {
        width: '80%',
        alignSelf: 'center',
        borderColor: 'grey',
        borderWidth: 1,
        marginBottom: 5,
        padding: 10
    },
    textAreaComment: {
        height: 80,
        width: '80%',
        alignSelf: 'center',
        borderColor: 'grey',
        borderWidth: 1,
        padding: 10
    },
    submitCommentBtn: {
        marginTop: 20,
        width: '80%',
        alignSelf: 'center'
    }
});

