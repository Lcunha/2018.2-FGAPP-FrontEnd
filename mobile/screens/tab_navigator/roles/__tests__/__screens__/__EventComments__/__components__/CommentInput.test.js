import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import CommentInput from '../../../../screens/EventComments/components/CommentInput';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<CommentInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
