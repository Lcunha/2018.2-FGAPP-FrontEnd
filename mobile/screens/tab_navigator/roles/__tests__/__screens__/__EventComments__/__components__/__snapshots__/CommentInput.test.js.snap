// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="always"
>
  <View>
    <TextInput
      allowFontScaling={true}
      maxLength={40}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Autor"
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderWidth": 1,
          "marginBottom": 5,
          "padding": 10,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      maxLength={128}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Novo Comentário"
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 80,
          "padding": 10,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#5cb85c",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "width": "80%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Enviar Comentário
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderWidth": 0.25,
          "margin": 20,
          "opacity": 0.25,
          "width": "80%",
        }
      }
    />
  </View>
</RCTScrollView>
`;
